backend/api/search.py

```diff
@@ -95,44 +79,67 @@ def search_avanced():
         res (json): JSON containing Projects and metadata
 
     """
-    text = request.args.get("q", type=str)
-    sorting = request.args.get('sort', type=str)
-    order = request.args.get('order', type=str)
-    offset = request.args.get('offset', type=int)
-    if offset is None:
-        offset = 0
-    count = request.args.get('count', type=int)
-    if count is None:
-        count = 10
-    save = request.args.get('save', type=str)
+
+    query = request.get_json()
+
+    try:
+        searchString = query['searchString']
+        authors = query.get('authors', [])
+        tags = query.get('tags', [])
+        status = query.get('status')
+        date_from = query.get('date_from')
+        date_to = query.get('date_to')
+        description = query.get('description')
+    except KeyError:
+        return make_response('Invalid json', 400)
 
     request_json = {
         'query': {
-            'query_string': {
-                'query': text,
+            'multi_match': {
+                'query': searchString,
+                'fields': [
+                    'tags^2',
+                    'title^2',
+                    'description',
+                ],
+            },
+            'filter': [
+                { 'term': { 'authors': authors } },
+                { 'term': { 'tags': tags } },
             },
         },
-        'from': offset,
-        'size': count,
     }
-    if (sorting is not None) and (order is not None):
-        request_json["sort"] = dict()
-        request_json["sort"][sorting] = {
-            'order': order,
+
+    if status:
+        request_json['query']['filter'].append({
+            'term': { 'status': status  }
+        })
+
+    if date_from or date_to:
+        date_filter = {
+            'range': { 'date_creation': {} }
         }
 
-    if save:
-        del request_json['from']
-        del request_json['size']
+        if date_from:
+            date_filter['range']['date_creation']['gte'] = date_from
+        if date_to:
+            date_filter['range']['date_creation']['lte'] = date_to
+
+        request_json['query']['filter'].append(date_filter)
+
+    if description:
+        request_json['query']['filter'].append({
+            'wildcard': { 'description': '*'+description+'*' }
+        })
```

Comment:
not really familiar with elasticsearch enough to comment on this properly, FE should test it before merge into master